src/string_buffer.rs

```diff
@@ -112,6 +122,17 @@ impl StringNode {
         }
     }
 
+    fn truncate(&mut self, new_len: usize) {
```

Comment:
this needs a comment to explain that new_len is relative to the start of this node, not the whole buffer